plugins/inputs/openntpd/openntpd.go

```diff
@@ -0,0 +1,254 @@
+	cmdArgs := []string{"-s", "peers"}
+
+	cmd := exec.Command(cmdName, cmdArgs...)
+
+	if UseSudo {
+		cmdArgs = append([]string{cmdName}, cmdArgs...)
+		cmd = exec.Command("sudo", cmdArgs...)
+	}
+
+	var out bytes.Buffer
+	cmd.Stdout = &out
+	err := internal.RunTimeout(cmd, Timeout.Duration)
+	if err != nil {
+		return &out, fmt.Errorf("error running ntpctl: %s", err)
+	}
+
+	return &out, nil
+}
+
+func (n *Openntpd) Gather(acc telegraf.Accumulator) error {
+	out, err := n.run(n.Binary, n.Timeout, n.UseSudo)
+	if err != nil {
+		return fmt.Errorf("error gathering metrics: %s", err)
+	}
+
+	// Due to problems with a parsing, we have to use regexp expression in order
+	// to remove string that starts from '(' and ends with space
+	// see: https://github.com/influxdata/telegraf/issues/2386
+	reg, err := regexp.Compile("\\s+\\([\\S]*")
+	if err != nil {
+		return err
+	}
+
+	lineCounter := 0
+	scanner := bufio.NewScanner(out)
+	for scanner.Scan() {
+		// skip first line
+		if lineCounter == 0 {
+			lineCounter++
+			continue
+		}
+
+		line := scanner.Text()
+
+		line = reg.ReplaceAllString(line, "")
+		fields := strings.Fields(line)
+
+		// second line contains field list ---> update mapping
+		if lineCounter == 1 {
+			for i, field := range fields {
+				// Check if field is a tag:
+				if tagKey, ok := tagHeaders[field]; ok {
+					tagI[tagKey] = i
+					continue
+				}
+
+				// check if field is a float metric:
+				if _, ok := floatI[field]; ok {
+					floatI[field] = i
+					continue
+				}
+
+				// check if field is an int metric:
+				if _, ok := intI[field]; ok {
+					intI[field] = i
+					continue
+				}
+			}
+			lineCounter++
+			continue
+		}
+
+		mFields := make(map[string]interface{})
+		tags := make(map[string]string)
+
+		// Even line ---> ntp server info
+		if lineCounter%2 == 0 {
+			tags["remote"] = fields[0]
+		}
+
+		// Read next line - Odd line ---> ntp server stats
+		scanner.Scan()
+		line = scanner.Text()
+		lineCounter++
+
+		line = reg.ReplaceAllString(line, "")
+		fields = strings.Fields(line)
+
+		// if there is an ntpctl state prefix, remove it and make it it's own tag
+		if strings.ContainsAny(string(fields[0]), "*#o+x.-") {
+			tags["state_prefix"] = string(fields[0])
+			fields = append(fields[:0], fields[1:]...)
+		}
+
+		// Get tags from output
+		for key, index := range tagI {
+			if index == -1 {
+				continue
+			}
+			tags[key] = fields[index]
+		}
+
+		// Get integer metrics from output
+		for key, index := range intI {
+			if index == -1 || index >= len(fields) {
+				continue
+			}
+			if fields[index] == "-" {
+				continue
+			}
+
+			if key == "next" || key == "poll" {
+
+				m, err := strconv.Atoi(strings.TrimSuffix(fields[index], "s"))
+				if err != nil {
+					acc.AddError(fmt.Errorf("E! Error ntpctl: parsing int: %s", fields[index]))
+					continue
+				}
+				mFields[key] = int64(m)
+
+			} else {
+
+				m, err := strconv.Atoi(fields[index])
```

Comment:
This probably doesn't matter here but always use ParseInt so there will be no funny business on 32-bit platforms.

Reply:
Switch to ParseInt done